### PR TITLE
Documentation for KubeVirt scheduling settings

### PR DESCRIPTION
**Pull request by hdurand0710**
Signed-off-by: Helene Durand <helene@kubermatic.com>

### CODE REVIEW

**Comment by hdurand0710** [comment]
Refer: Kubermatic/kubermatic#10724

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 1]
```diff
@@ -138,6 +138,279 @@ All the resources related to VM on the KubeVirt cluster will be created in a ded
...
+![Scheduling](./Scheduling.png)
+
+We provide 3 different types of scheduling for the KubeVirt tenant nodes:
+- Ensure co-location on the same infrastructure node (*Pod Affinity Preset*).
+- Prevent co-locotation on the same infrastructure node (*Pod Anti Affinity Preset*).
+- Schedule on nodes having some specific labels (*Node Affinity Preset*).
+
+This setup is done in the `Initial Nodes` step of KKP dashboard when creating a cluster.
+
+![Dashboard](./Dashboard-scheduling.png)
```
Comment (Waseem826) [inline content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:154]: The UI for Affinity Settings has been updated in https://github.com/kubermatic/dashboard/pull/4813 so you might want to update this screenshot once that is merged. 

Here's how it looks now:

screenshot-localhost_8000-2022 08 11-22_21_30

Comment (hdurand0710) [reply content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:152]: Screenshot is updated.

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 2]
```diff
@@ -138,6 +136,279 @@ All the resources related to VM on the KubeVirt cluster will be created in a ded
...
 
+### Advanced scheduling configuration
+
+#### Scheduling settings
+It's possible to control how the tenant nodes are scheduled on the infrastructure nodes.
+![Scheduling](./Scheduling.png)
+
+We provide 3 different types of scheduling for the KubeVirt tenant nodes:
+- Ensure co-location on the same infrastructure node (*Pod Affinity Preset*).
+- Prevent co-locotation on the same infrastructure node (*Pod Anti Affinity Preset*).
```
Comment (sankalp-r) [inline content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:147]: Typo `co-locotation` --> `co-location`

Comment (hdurand0710) [reply content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md]: Fixed

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 3]
```diff
@@ -167,6 +169,276 @@ For each of them, the user must specify:
...
+
+
+*"hard"* or *"soft"* are related to the [Kubernetes: Assigning Pods to Nodes](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/). The following table shows the mapping:
+| Affinity type | Kubernetes affinity                             |
+|---------------|-------------------------------------------------|
+| hard          | requiredDuringSchedulingIgnoredDuringExecution  |
+| soft          | preferredDuringSchedulingIgnoredDuringExecution |
+
+
+#### How to set up the scheduling
```
Comment (mfranczy) [inline content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:201]: I would change to:
> How scheduling settings influence a MachineDeployment object

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 4]
```diff
@@ -167,6 +169,276 @@ For each of them, the user must specify:
...
+
+This set up is defined in the *MachineDeployment* `spec.affinity`:
```
Comment (mfranczy) [inline content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:203]: I would change to
> Scheduling settings are represented in the `MachineDeployment` object under `spec.affinity`:

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 5]
```diff
@@ -167,6 +169,276 @@ For each of them, the user must specify:
...
+              nodeAffinityPreset:
+                type: "" # Allowed values: "", "soft", "hard"
+                key: "foo"
+                values:
+                  - bar
+// ....
+```
+
+**Important Note**:
+- `podAffinityPreset` and `podAntiAffinityPreset` are **mutually exclusive**. It does not make sense to schedule both on a single infrastructure node and prevent from being on the same infrastructure node.
```
Comment (mfranczy) [inline content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:226]: I would change to:
> `podAffinityPreset` and `podAntiAffinityPreset` are **mutually exclusive**. The Anti-Affinity setting works the opposite of Affinity.

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 6]
```diff
@@ -167,6 +169,276 @@ For each of them, the user must specify:
...
+- `podAffinityPreset` can be specified along with `nodeAffinityPreset`: this allows ensure that the KubeVirt tenant nodes are co-located on a single infrastructure node that has some specific labels.
```
Comment (mfranczy) [inline content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:227]: I would change to
> on a single infrastructure node that has the specified labels (read how to add node labels)

We would have to add a link to part of the documentation that described how to specify labels in the UI.

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md [hunk 7]
```diff
@@ -167,6 +169,276 @@ For each of them, the user must specify:
...
+- `podAntiAffinityPreset` can be specified along with `nodeAffinityPreset`: this prevents the KubeVirt tenant nodes from being co-located on the same infrastructure node, but be located on infrastructure nodes that have some specific labels.
```
Comment (mfranczy) [inline content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:228]: Similar like podAffinityPreset

**Review by mfranczy — COMMENTED** [review]
Overall looks great, I have a few small comments.

Could we re-work a bit the `Scheduling.png`? It's not clear that KubeVirt logo represents a tenant node. Even with the legend on the picture. Since we use description text for bare-metal nodes then I would do the same for tenant nodes. In the box I would add another boxes with different color and the description `Tenant node N`.

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md (hunk 6 above)
Comment (hdurand0710) [reply content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:227]: We can not label infrastructure nodes with the UI right ?

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md (hunk 3 above)
Comment (hdurand0710) [reply content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:201]: done

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md (hunk 4 above)
Comment (hdurand0710) [reply content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:203]: done

File: content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md (hunk 6 above)
Comment (mfranczy) [reply content/kubermatic/master/architecture/supported-providers/kubevirt/kubevirt.en.md:227]: you got me :) Nevermind then.

**Review by mfranczy — APPROVED** [review]
Nice!
/lgtm

**Pull request merged**